python_modules/dagster-graphql/dagster_graphql/schema/asset_health.py

```diff
@@ -0,0 +1,78 @@
+import graphene
+
+from dagster_graphql.schema.util import ResolveInfo
+
+
+class GrapheneAssetHealthStatus(graphene.Enum):
+    HEALTHY = "HEALTHY"
+    WARNING = "WARNING"
+    DEGRADED = "DEGRADED"
+    UNKNOWN = "UNKNOWN"
+
+    class Meta:
+        name = "AssetHealthStatus"
+
+
+class GrapheneAssetHealth(graphene.ObjectType):
+    assetHealth = graphene.NonNull(GrapheneAssetHealthStatus)
+    materializationStatus = graphene.Field(GrapheneAssetHealthStatus)
+    assetChecksStatus = graphene.Field(GrapheneAssetHealthStatus)
+    freshnessStatus = graphene.Field(GrapheneAssetHealthStatus)
+
+    class Meta:
+        name = "AssetHealth"
+
+    def resolve_assetHealth(self, graphene_info: ResolveInfo):
+        if not graphene_info.context.instance.asset_health_supported():
+            return GrapheneAssetHealthStatus.UNKNOWN
+        statuses = [
+            self.resolve_materializationStatus(graphene_info),
+            self.resolve_assetChecksStatus(graphene_info),
+            self.resolve_freshnessStatus(graphene_info),
+        ]
+        if GrapheneAssetHealthStatus.DEGRADED in statuses:
+            return GrapheneAssetHealthStatus.DEGRADED
+        if GrapheneAssetHealthStatus.WARNING in statuses:
+            return GrapheneAssetHealthStatus.WARNING
+        if all(status == GrapheneAssetHealthStatus.UNKNOWN for status in statuses):
+            return GrapheneAssetHealthStatus.UNKNOWN
+        return GrapheneAssetHealthStatus.HEALTHY
+
+    def resolve_materializationStatus(self, graphene_info: ResolveInfo):
+        # if non-partitioned:
+        # if latest materialization succeed, healthy
+        # if latest materialization failed, degraded
+        # no materialization, unknown
+        # if partitioned:
+        # if all partitions successfully materialized, healthy
+        # if any partitions missing, warning
+        # if any partitions failed, degraded
+        if (
+            not graphene_info.context.instance.asset_health_supported()
+            or not graphene_info.context.instance.can_read_failure_events()
```

Comment:
we need a more complex check for status since the failure events may not be being emitted (it's controlled by different feature flags)